asr-noise-augmentation.ipynb

```diff
@@ -0,0 +1,605 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Training dataset\n",
+    "Let's create a noise-augmented training dataset using the AN4 training dataset. We'll add noise at different SNRs (Signal-to-Noise Ratios) ranging from 0 to 15 dB SNR using a NeMo script. Note that a 0 dB SNR means that the noise and signal in the given audio file are of equal volume. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "final_data_dir = os.path.join(data_dir, 'noise_data')\n",
+    "\n",
+    "train_manifest = os.path.join(data_dir, 'an4/train_manifest.json')\n",
+    "test_manifest  = os.path.join(data_dir, 'an4/test_manifest.json')\n",
+    "\n",
+    "train_noise_manifest = os.path.join(data_dir, 'train_noise_manifest.json')\n",
+    "test_noise_manifest  = os.path.join(data_dir, 'test_noise_manifest.json')\n",
+    "\n",
+    "!python $nemo_dir/scripts/dataset_processing/add_noise.py \\\n",
+    "    --input_manifest=$train_manifest \\\n",
+    "    --noise_manifest=$train_noise_manifest \\\n",
+    "    --snrs 0 5 10 15 \\\n",
+    "    --out_dir=$final_data_dir"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above script generates a `.json` manifest file each for every SNR value, i.e., one manifest file each for 0, 5, 10 and 15db SNR. \n",
+    "\n",
+    "First, let's give these manifest files less cumbersome names."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "noisy_train_manifest_files = os.listdir(os.path.join(final_data_dir, 'manifests'))\n",
+    "for filename in noisy_train_manifest_files:\n",
+    "    new_filename = filename.replace('train_manifest_train_noise_manifest', 'noisy_train_manifest')\n",
+    "    new_filepath = os.path.join(final_data_dir, 'manifests', new_filename)\n",
+    "    filepath = os.path.join(final_data_dir, 'manifests', filename)\n",
+    "    os.rename(filepath, new_filepath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's combine all the manifests for noise-augmented training data into a single manifest."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat $final_data_dir/manifests/noisy* > $final_data_dir/manifests/noisy_train_manifest.json\n",
+    "\n",
+    "print(\"Combined manifest for noise-augmented training dataset created at\", final_data_dir + \"/manifests/noisy_train_manifest.json\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Test dataset\n",
+    "\n",
+    "Let's create a noise-augmented evaluation dataset using the an4 test dataset, by adding noise at 5 dB, using the same NeMo `add_noise.py` script with which we augmented the training dataset."
```

Comment:
Its an4 and AN4 throughout this tutorial. Can we be consistent with just one naming style?

Reply:
I made a few changes, which I'll commit and push momentarily. It should now be consistently rendered "AN4" in Markdown cells and printouts and "an4" in variable names and file paths.